File: examples/mixed-example/src/main/kotlin/nz/bradcampbell/mixedexample/State.java
package nz.bradcampbell.mixedexample;

import com.google.auto.value.AutoValue;
import android.os.Parcelable;

import java.util.Date;

@AutoValue
public abstract class State implements Parcelable {
  public abstract Test count();
  public abstract Date modificationDate();

  public static State create(Test count, Date modificationDate) {
    return new AutoValue_State(count, modificationDate);
  }
}

File: kotlin-example/src/main/kotlin/nz/bradcampbell/paperparcel/kotlinexample/State.kt
package nz.bradcampbell.paperparcel.kotlinexample

import nz.bradcampbell.paperparcel.PaperParcel
import java.util.*

@PaperParcel(typeAdapters = arrayOf(DateTypeAdapter::class))
data class State(val count: Int, val modificationDate: Date)
